service/header/exchange_test.go

```diff
@@ -0,0 +1,254 @@
+}
+
+// TestExchange_Response_Single tests that the exchange instance can respond
+// to a ExtendedHeaderRequest for one ExtendedHeader accurately.
+func TestExchange_Response_Single(t *testing.T) {
+	net, err := mocknet.FullMeshConnected(context.Background(), 2)
+	require.NoError(t, err)
+	// get host and peer
+	host, peer := net.Hosts()[0], net.Hosts()[1]
+	// fill out mockstore
+	store := new(mockStore)
+	store.headers = make(map[int]*ExtendedHeader)
+	for i := 1; i <= 5; i++ {
+		store.headers[i] = RandExtendedHeader(t)
+		store.headers[i].Height = int64(i)
+	}
+	// create exchange just to register the stream handler
+	_ = newExchange(host, libhost.InfoFromHost(peer), store)
+	// start a new stream via Peer to see if Host can handle inbound requests
+	stream, err := peer.NewStream(context.Background(), libhost.InfoFromHost(host).ID, headerExchangeProtocolID)
+	require.NoError(t, err)
+	// create request
+	origin := uint64(3)
+	req := &ExtendedHeaderRequest{
+		Origin: origin,
+		Amount: 1,
+	}
+	// send request
+	_, err = serde.Write(stream, req.ToProto())
+	require.NoError(t, err)
+	// read resp
+	resp := new(header_pb.ExtendedHeader)
+	_, err = serde.Read(stream, resp)
+	require.NoError(t, err)
+	// compare
+	got, err := ProtoToExtendedHeader(resp)
+	require.NoError(t, err)
+	assert.Equal(t, store.headers[int(origin)].Height, got.Height)
+	assert.Equal(t, store.headers[int(origin)].Hash(), got.Hash())
+}
+
+// TestExchange_Response_Multiple tests that the exchange instance can respond
+// to a ExtendedHeaderRequest for multiple ExtendedHeaders accurately.
+func TestExchange_Response_Multiple(t *testing.T) {
+	net, err := mocknet.FullMeshConnected(context.Background(), 2)
+	require.NoError(t, err)
+	// get host and peer
+	host, peer := net.Hosts()[0], net.Hosts()[1]
+	// fill out mockstore
+	store := new(mockStore)
+	store.headers = make(map[int]*ExtendedHeader)
+	for i := 1; i <= 5; i++ {
+		store.headers[i] = RandExtendedHeader(t)
+		store.headers[i].Height = int64(i)
+	}
+	// create exchange just to register the stream handler
+	_ = newExchange(host, libhost.InfoFromHost(peer), store)
+	// start a new stream via Peer to see if Host can handle inbound requests
+	stream, err := peer.NewStream(context.Background(), libhost.InfoFromHost(host).ID, headerExchangeProtocolID)
+	require.NoError(t, err)
+	// create request
+	origin := uint64(3)
+	req := &header_pb.ExtendedHeaderRequest{
+		Origin: origin,
+		Amount: 2,
+	}
+	// send request
+	_, err = serde.Write(stream, req)
+	require.NoError(t, err)
+	// read responses
+	for i := origin; i < (origin + req.Amount); i++ {
+		resp := new(header_pb.ExtendedHeader)
+		_, err := serde.Read(stream, resp)
+		require.NoError(t, err)
+		eh, err := ProtoToExtendedHeader(resp)
+		require.NoError(t, err)
+		// compare
+		assert.Equal(t, store.headers[int(i)].Height, eh.Height)
+		assert.Equal(t, store.headers[int(i)].Hash(), eh.Hash())
+	}
+}
+
+type mockStore struct {
```

Comment:
I changed the Store interface a bit in the Store PR, so we should merge it first and slightly change this mock structure.